machinelearning/caffe/README.md

```diff
@@ -0,0 +1,220 @@
+
+2018-10-23 14:51:40 (426 KB/s) - 't10k-images-idx3-ubyte.gz' saved [1648877/1648877]
+
+--2018-10-23 14:51:40--  http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz
+Resolving yann.lecun.com (yann.lecun.com)... 216.165.22.6
+Connecting to yann.lecun.com (yann.lecun.com)|216.165.22.6|:80... connected.
+HTTP request sent, awaiting response... 200 OK
+Length: 4542 (4.4K) [application/x-gzip]
+Saving to: 't10k-labels-idx1-ubyte.gz'
+
+t10k-labels-idx1-ubyte.gz                             100%[======================================================================================================================>]   4.44K  --.-KB/s    in 0.007s  
+
+2018-10-23 14:51:40 (652 KB/s) - 't10k-labels-idx1-ubyte.gz' saved [4542/4542]
+
+```
+
+So now all the downloaded data sets can be seen in `caffehost/data/mnist` folder.
+
+Now, before running the `create_mnist.sh` script located in `caffehost/examples/mnist`, we will need first to modify the paths in there. To do this, open the `create_mnist.sh` on your favorite editor and edit the following 3 lines inside the script:
+
+```
+EXAMPLE=/caffehost/examples/mnist
+DATA=/caffehost/data/mnist
+BUILD=/caffehost/build/examples/mnist
+```
+
+With this configuration, now go to `/examples/mnist` and run:
+
+```
+./create_mnist.sh
+```
+ You should obtain an output similar to this:
+
+ ```
+ Singularity caffe.sif:/caffehost/examples/mnist> ./create_mnist.sh
+ Creating lmdb...
+ I1023 15:48:16.656437 14117 db_lmdb.cpp:35] Opened lmdb /caffehost/examples/mnist/mnist_train_lmdb
+ I1023 15:48:16.656642 14117 convert_mnist_data.cpp:88] A total of 60000 items.
+ I1023 15:48:16.656651 14117 convert_mnist_data.cpp:89] Rows: 28 Cols: 28
+ I1023 15:48:21.751487 14117 convert_mnist_data.cpp:108] Processed 60000 files.
+ I1023 15:48:22.010707 14134 db_lmdb.cpp:35] Opened lmdb /caffehost/examples/mnist/mnist_test_lmdb
+ I1023 15:48:22.010915 14134 convert_mnist_data.cpp:88] A total of 10000 items.
+ I1023 15:48:22.010923 14134 convert_mnist_data.cpp:89] Rows: 28 Cols: 28
+ I1023 15:48:22.814141 14134 convert_mnist_data.cpp:108] Processed 10000 files.
+ Done.
+ ```
+
+#### LeNet: the MNIST Classification Model
+
+To run the training program, the [LeNet](http://yann.lecun.com/exdb/publis/pdf/lecun-01a.pdf) network will be used. Its use is known in digital classification tasks. For this specific case, the LeNet implementation will be changed replacing the sigmoid activations with Rectified Linear Unit (ReLU) activations for the neurons.
+
+The layers are defined inside `caffe/examples/mnist/lenet_train_test.prototxt`.
+
+
+#### Defining the MNIST network
+
+We will explain the `lenet_train_test.prototxt` model. The format is as much as similar as from [Google Protobuf](https://developers.google.com/protocol-buffers/docs/overview) also, the definitions regarding protobuf from caffe can be found in `caffe/src/caffe/proto/caffe.proto`.
```

Comment:
`We will explain the `lenet_train_test.prototxt` model.`

Err? This section should probably be reworded ... @GodloveD , something like? :
```
We will be using the `lenet_train_test.prototxt` model. This model is similar to [Google Protobuf]
(https://developers.google.com/protocol-buffers/docs/overview). You can find the definitions in: 
`caffe/src/caffe/proto/caffe.proto`.
```